cmd/apprepository-controller/server/controller.go

```diff
@@ -354,541 +403,167 @@ func (c *Controller) syncHandler(key string) error {
-func apprepoSyncJobEnvVars(apprepo *apprepov1alpha1.AppRepository, config Config) []corev1.EnvVar {
-	var envVars []corev1.EnvVar
-	envVars = append(envVars, corev1.EnvVar{
-		Name: "DB_PASSWORD",
-		ValueFrom: &corev1.EnvVarSource{
-			SecretKeyRef: &corev1.SecretKeySelector{
-				LocalObjectReference: corev1.LocalObjectReference{Name: config.DBSecretName},
-				Key:                  config.DBSecretKey,
-			},
-		},
-	})
-	if apprepo.Spec.Auth.Header != nil {
-		if apprepo.Spec.Auth.Header.SecretKeyRef.Key == ".dockerconfigjson" {
-			envVars = append(envVars, corev1.EnvVar{
-				Name: "DOCKER_CONFIG_JSON",
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: secretKeyRefForRepo(apprepo.Spec.Auth.Header.SecretKeyRef, apprepo, config),
-				},
-			})
+	// if the object is being deleted and contains a finalizer
+	if !apprepo.GetDeletionTimestamp().IsZero() && containsFinalizer(apprepo, finalizerName) {
+		// if the object is being deleted and contains a finalizer and the event is not a deletion event,
+		// then handle the finalizer-derived clean-up tasks and remove the finalizer
+		if !shouldDelete {
+			err = c.handleCleanupFinalizer(ctx, apprepo)
+			if err != nil {
+				log.Errorf("Error handling the finalizer of the AppRepository %q: %v", apprepoKey, err)
+				return
+			}
 		} else {
-			envVars = append(envVars, corev1.EnvVar{
-				Name: "AUTHORIZATION_HEADER",
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: secretKeyRefForRepo(apprepo.Spec.Auth.Header.SecretKeyRef, apprepo, config),
-				},
-			})
+			// if the object is being deleted and it is a deletion event, then do nothing else
+			log.Infof("The AppRepository %q is now being deleted", apprepoKey)
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err == nil {
+				c.workqueue.AddRateLimited(key)
+			}
 		}
 	}
-	return envVars
-}
-
-// secretKeyRefForRepo returns a secret key ref with a name depending on whether
-// this repo is in the kubeapps namespace or not. If the repo is not in the
-// kubeapps namespace, then the secret will have been copied from another namespace
-// into the kubeapps namespace and have a slightly different name.
-func secretKeyRefForRepo(keyRef corev1.SecretKeySelector, apprepo *apprepov1alpha1.AppRepository, config Config) *corev1.SecretKeySelector {
-	if apprepo.ObjectMeta.Namespace == config.KubeappsNamespace {
-		return &keyRef
-	}
-	keyRef.LocalObjectReference.Name = helm.SecretNameForNamespacedRepo(apprepo.ObjectMeta.Name, apprepo.ObjectMeta.Namespace)
-	return &keyRef
-}
-
-// apprepoCleanupJobArgs returns a list of args for the repo cleanup container
-func apprepoCleanupJobArgs(namespace, name string, config Config) []string {
-	return append([]string{
-		"delete",
-		name,
-		"--namespace=" + namespace,
-	}, dbFlags(config)...)
-}
-
-func dbFlags(config Config) []string {
-	return []string{
-		"--database-url=" + config.DBURL,
-		"--database-user=" + config.DBUser,
-		"--database-name=" + config.DBName,
-	}
 }
```

Comment:
Just noting that there's a lot of deletions in this file - guessing they'll be re-added below. EDIT: yep, see them in the job_utils. Mild preference for keeping moves of code like this to separate PRs, but realise it's not always possible when the code is changing too. Sometimes I do the code changes and refrain from moving/reorganising until a follow-up PR, just so it's easier to follow what has actually changed.